### PR TITLE
Monitoring - Sites Overview

### PR DESCRIPTION
**Pull request by echaidemenos**
Screenshot from 2023-12-12 16-50-35

### CODE REVIEW

File: packages/api/src/monitoring/dto/get-sites-overview.dto.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  @ApiProperty({ example: 42 })
+  @IsOptional()
+  @IsNumber()
+  @Validate(EntityExists, [Site])
+  siteId?: number;
+
+  @ApiProperty({ example: 'Bellows South Africa' })
+  @Type(() => String)
+  @IsOptional()
+  siteName?: string;
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply packages/api/src/monitoring/dto/get-sites-overview.dto.ts:23]: I added validation about the email format.
About `siteName` and `fullName`, there is no limit on the string size when creating them, so having a limit here on an admin use only endpoint, might not be what we want. Maybe we should add limits when creating also. what do you think @ericboucher ?

[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/api/src/monitoring/dto/get-sites-overview.dto.ts:23]: Yes let's just add limits when creating

File: packages/api/src/monitoring/monitoring.service.ts [hunk 2]
```diff
@@ -230,4 +235,105 @@ export class MonitoringService {
...
+      .orderBy('latest_data.site_id')
+      .addOrderBy('latest_data.timestamp', 'DESC');
+
+    const surveysCountSubQuery = this.surveyRepository
+      .createQueryBuilder('survey')
+      .select('survey.site_id', 'site_id')
+      .addSelect('COUNT(*)', 'surveysCount')
+      .groupBy('survey.site_id');
+
+    const baseQuery = this.siteRepository
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply packages/api/src/monitoring/monitoring.service.ts:263]: I did some test runs and the query is complete in less than a second, in production. There may be room for improvement, but for the scale of the data and the usage, i believe is good for now :)

File: packages/website/src/routes/MonitoringRoutes/SitesOverview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      label: 'Admin Username',
+      value: adminUserName,
+      setValue: setAdminUserName,
+    },
+  ];
+
+  const filters = (
+    <div className={classes.filtersWrapper}>
+      {textFilters.map((elem) => (
+        <TextField
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/website/src/routes/MonitoringRoutes/SitesOverview/index.tsx:81]: @echaidemenos still outstanding?

**Review by ericboucher — APPROVED** [review]

**Pull request merged**